src/System.Net.HttpListener/tests/HttpRequestStreamTests.cs

```diff
@@ -505,13 +505,10 @@ public void Dispose()
                 byte[] buffer = new byte[expected.Length];
                 await Assert.ThrowsAsync<HttpListenerException>(() => context.Request.InputStream.ReadAsync(buffer, 0, buffer.Length));
                 await Assert.ThrowsAsync<HttpListenerException>(() => context.Request.InputStream.ReadAsync(buffer, 0, buffer.Length));
-
-                // Closing a response from a closed client if no writing has failed should fail.
-                Assert.Throws<HttpListenerException>(() => context.Response.Close());
```

Comment:
Because it seemed like weird behavior.  The client has sent a malformed request; once I get the exception on reading the request, what else am I supposed to do except Close?  I would understand if Write threw, but Close seems weird.

If this is Windows behavior, then maybe we should consider whether that should change.

Thoughts?